downloads/swan-lake-release-notes/swan-lake-2201.7.0/RELEASE_NOTE.md

```diff
@@ -27,6 +27,29 @@ Update your current Ballerina installation directly to 2201.7.0 by using the [Ba
 If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.
 
 ## Backward-incompatible changes
+- Fixed a bug that sets broad type for constants instead of singleton type.
+
+  ```ballerina
+  const int CONST = 1 + 7;
+  CONST num = 2;
+  ```
+  Now, the code produces an error: `incompatible types: expected '8', found 'int'`. Similarly, the following assignment in the function scope also results in an error: `incompatible types: expected 'SingletonType', found 'int'`.
+
+  ```ballerina
+  const int CONST = 3 + 5;
+  type SingletonType CONST;
+  
+  public function main() {
+    SingletonType num = 3;
+  }
+  ```
+  
+- Fixed a bug that caused a runtime error instead of a compile-time error for constant declarations containing duplicate keys in computed fields in mapping constructor expressions.
```

Comment:
```suggestion
- A bug that resulted in compilation errors not being logged for duplicate keys via computed name fields of a mapping constructor in a map constant declaration has been fixed. This previously resulted in a runtime panic instead.
```

Reply:
Yes it should result in compilation error. I will create an issue.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/40727